server/cloudformation/environment.go

```diff
@@ -0,0 +1,145 @@
+package cloudformation
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/remind101/empire"
+	"golang.org/x/net/context"
+)
+
+// envClient mocks the Empire interface we use.
+type envClient interface {
+	AppsFind(empire.AppsQuery) (*empire.App, error)
+	Set(context.Context, empire.SetOpts) (*empire.Config, error)
+}
+
+type Variable struct {
+	Name  *string
+	Value *string
+}
+
+// EnvironmentProperties represents the properties for the
+// Custom::EmpireAppEnvironment
+type EnvironmentProperties struct {
+	AppId     *string
+	Variables []Variable
+}
+
+// EnvironmentResource is a Provisioner that manages environmental variables
+// within an Empire application.
+type EnvironmentResource struct {
+	empire envClient
+}
+
+func (p *EnvironmentResource) Properties() interface{} {
+	return &EnvironmentProperties{}
+}
+
+type VariableError struct {
+	index int
+	err   string
+}
+
+func (v *VariableError) Error() string {
+	return fmt.Sprintf("invalid variable [%d]: %s", v.index, v.err)
+}
+
+func (p *EnvironmentResource) Provision(req Request) (id string, data interface{}, err error) {
+	ctx := context.Background()
+	user := NewUser()
+	properties := req.ResourceProperties.(*EnvironmentProperties)
+
+	switch req.RequestType {
+	case Create:
+		if *properties.AppId == "" {
+			return "", nil, fmt.Errorf("missing parameter: AppId")
+		}
+		id = *properties.AppId
```

Comment:
from the docs:

> This value should be an **identifier unique to the custom resource vendor**, and can be up to 1Kb in size. The value must be a non-empty string.

i believe both of these are treated as different resource vendors, so the PhysicalResourceId should just need to be unique within that scope.